### DISCUSSION TITLE
为什么不一个jfs共享一个aes key?

### DISCUSSION THREAD

**Discussion by meilihao** [juicedata/juicefs#485] [Q&A] [answered]
当前每个数据块一个aes key, 还要用rsa 加密, 这样即增大了数据块的大小也因rsa加密拖累了性能.

**Comment by davies** [comment, marked as answer]
如果使用单一的 AES 密钥，因为它的加解密速度快，有被暴力破解的风险，一旦密钥泄漏，所有的数据都泄露了。目前的两层结构，每个对象都使用随机生成的  AES 密钥，几遍被暴力破解了也不会泄漏所有数据。而 RSA 的加解密很慢，被暴力破解的风险低很多。

从我们之间的测试来看，RSA 只用于加解密 AES 密钥，对 CPU 消耗和性能的影响不大，你可以实测一下看看真实场景的影响。

我们也欢迎增加新的加密算法，将它变成可配置的。

**Reply by meilihao** [reply]
明白了. 我对加密的理解一直来源于https/tls, 它只在密钥交换时使用了rsa, 通信过程中仅使用了aes.
